x-pack/test/security_solution_cypress/cypress/e2e/investigations/timeline_templates/creation.cy.ts

```diff
@@ -43,79 +39,103 @@ import {
-
-      cy.log('Check that the template has been created correclty');
-      cy.contains(getTimeline().title).should('exist');
-      cy.get(TIMELINE_TITLE).should('have.text', getTimeline().title);
-      cy.get(TIMELINES_DESCRIPTION).first().should('have.text', getTimeline().description);
-      cy.get(TIMELINES_FAVORITE).first().should('exist');
-      cy.get(TIMELINE_QUERY).should('contain.text', getTimeline().query);
+
+      cy.get(TIMELINE_TITLE_BY_ID(savedObjectId)).should('have.text', mockTimeline.title);
+      cy.get(TIMELINES_DESCRIPTION).first().should('have.text', mockTimeline.description);
+      cy.get(TIMELINES_USERNAME).first().should('have.text', createdBy);
+
+      cy.log('Open template and check that the template has been created correctly');
+
+      openTimelineTemplate(savedObjectId);
+
+      cy.get(TIMELINE_TITLE).should('have.text', mockTimeline.title);
+      cy.get(TIMELINE_QUERY).should('contain.text', mockTimeline.query);
+      cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM_AT(0))
+        .should('contain.text', mockTimeline.filter.field)
+        .and('contain.text', mockTimeline.filter.value);
+      cy.get(
+        GET_LOCAL_DATE_PICKER_START_DATE_POPOVER_BUTTON(TIMELINE_DATE_PICKER_CONTAINER)
+      ).should('have.text', startDate);
+      cy.get(GET_LOCAL_DATE_PICKER_END_DATE_POPOVER_BUTTON(TIMELINE_DATE_PICKER_CONTAINER)).should(
+        'not.have.text',
+        'now'
+      );
     });
   });
 
-  it('Create template from timeline', () => {
-    createTimeline(getTimeline());
+  it('should create a template from an existing timeline', () => {
+    createTimeline(mockTimeline);
     visit(TIMELINES_URL);
     waitForTimelinesPanelToBeLoaded();
     expandEventAction();
     clickingOnCreateTemplateFromTimelineBtn();
-    addNameToTimelineAndSave('Test');
-    cy.wait('@timeline', { timeout: 100000 });
-    cy.get(TIMELINE_FLYOUT_WRAPPER).should('have.css', 'visibility', 'visible');
-    cy.get(TIMELINE_QUERY).should('have.text', getTimeline().query);
-  });
+    const savedName = 'Test';
+    addNameToTimelineAndSave(savedName);
 
-  it('should create timeline template from bottombar', () => {
-    visit(TIMELINES_URL);
-    createTimelineTemplateFromBottomBar();
-    cy.get(TIMELINE_TITLE).should('have.text', 'Untitled template');
+    cy.wait('@timeline').then(({ response }) => {
+      const { createdBy, savedObjectId } = response?.body.data.persistTimeline.timeline;
+
+      cy.log('Check that the template has been created correctly');
+
+      cy.get(TIMELINE_TITLE).should('have.text', savedName);
+      cy.get(TIMELINE_QUERY).should('have.text', mockTimeline.query);
+
+      cy.log('Close timeline and verify template shows on the table in the templates tab');
+
+      closeTimeline();
+      openTimelineTemplatesTab();
+
+      cy.get(TIMELINE_TITLE_BY_ID(savedObjectId)).should('have.text', savedName);
+      cy.get(TIMELINES_DESCRIPTION).first().should('have.text', mockTimeline.description);
```

Comment:
Not necessary but man, `TIMELINES_DESCRIPTION` === `description` may be the least specific selector ever... 😓

Reply:
I know... If you look at this [timelines.ts](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_cypress/cypress/screens/timelines.ts) file it's full of them. Pretty much ALL those selector are the most generic ever 😢 😭